### ISSUE TITLE
Support Google login for users on Android

### ISSUE THREAD

**Issue by ryanmerkley** [creativecommons/list#140]
I'm filing this here to ensure we have a discussion about supporting Google login for users on Android, as Ben Adida suggested in his review of the app. Please do not close this item or mark as WON'T until we've had the discussion with Ben.

**Comment by ryanmerkley** [comment]
Ben's comments provided to Playground:

"If you’re targeting Android first, you should probably allow for using the pre-loaded Google identities on the phone (and making login optional in any case.) The overhead for users creating an account on mobile is very high. One approach is Auth0.com, which does an interesting login widget. Again, the goal of my suggestion here is simply to let you focus on the stuff that’s unique to CC, not the generic platform aspects that others can do better / more cheaply."

**Comment by perpetuatingcuriosity** [comment]
@ryanmerkley So this week I’m working on #92. Our plan was to use Google AccountManager and have CCID be the method by which users authenticate. If we ask for the appropriate permissions in the app, the user's CCID account gets automatically created and stored here: ![account_add_from_setting](https://cloud.githubusercontent.com/assets/6563209/6028055/84269cd6-abb4-11e4-98de-6f1156cf97c0.png)

I think what Ben is saying is that we shouldn't be spending our time on any of the authentication stuff at all. We could use the Auth0.com widget (which looks really cool and straightforward to use) and then we could focus our time and energy on the features that are unique to The List.

But the catch, if I understand correctly, is that this would replace our use of CCID (i.e.: there would be a standard username/password managed by Auth0 but we could also offer Facebook, Google+, Twitter, etc., to login) . Unless, of course, there is a way to integrate CCID with the Auth0 framework. But this might be creating more work for ourselves after all. @matll what do you think? Is there a way to get CCID working with Auth0? Alternately, are we comfortable not using CCID at all? Or should we stick with the original plan? We did get the session tokens working last week after all…

**Comment by mattl** [comment]
Yeah, I think we should continue with CCID. We've already got that working nicely with the app.
CCID may at some point use Google, Twitter, etc, but that'll be invisible to The List app. 

And you have the optional login already working, so maybe the app shouldn't even ask for a login at first, but instead just drops you into the app?

**Comment by mattl** [comment]
Also, @perpetuatingcuriosity's work would make the CCID available to other apps that want to integrate.

**Comment by perpetuatingcuriosity** [comment]
@ryanmerkley Can we close this? Currently we have CCID accounts being created and stored on the user’s phone: 

![screen shot 2015-02-09 at 11 40 37 am](https://cloud.githubusercontent.com/assets/6563209/6110560/815f1564-b050-11e4-93fd-f39571fe919b.png)

woohoo :dancer:

**Comment by mattl** [comment]
https://github.com/creativecommons/CCID-cas/issues/3

**Comment by benadida** [comment]
Hi @mattl @perpetuatingcuriosity,

I recommend reopening this issue. It is true that maintaining a login system is a significant endeavor on its own, and just getting it working once is far from all the work you're signing up for. But my point is less about development time to build CCID, and more about the user experience. Creating a custom login is very significant friction for users, especially on mobile. If you're going to add that friction, there has to be significant value, and that value has to be obvious to users. I'm not seeing anything compelling on either front.

**Comment by ryanmerkley** [comment]
I'm reopening it, and setting the milestone to v1.0. This isn't a conversation we have to have immediately, but I think we have to be deliberate about it, and I want to consider the issues Ben is raising (which was the reason I filed the issue in the first place). Thanks.

**Comment by mattl** [comment]
Moved to after demo day release.

Hey @benadida -- the current plan is to support all those other logins for CCID, rather than just one application.

**Comment by benadida** [comment]
@mattl good to hear!

I would suggest that the key issue to keep in mind is to drive everything from the user experience. There's a risk, if you start with components you think make sense logically and build up to a user experience, that the user experience ends up not making sense for someone who hasn't thought through the components the way you have.

So I suggest keeping in mind this one question: what is the easiest thing for the user? Build towards that ASAP, and leave future features to the future.

**Comment by ryanmerkley** [comment]
Comments from David Ascher (Mozilla) after reviewing the app (shared w/ permission):

"the flow from the app to the website to email to the app is way too complicated.  I expect you'll lose 90% of people in the account creation flow.  We're currently rethinking our account sign up / login experience, will be happy to share where we end up.  If you're doing Android only, I'd very much suggest offering a Google Auth as an option, it would be way smoother."

**Comment by mattl** [comment]
Added a note to CCID to do more on this.
